File: server/src/main/java/ru/vitaliy/petrov/server/forms/requests/documents/UserAccidentDocumentAddRequest.java
package ru.vitaliy.petrov.server.forms.requests.documents;

import lombok.Data;

import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;
import java.sql.Date;
import java.sql.Time;

@Data
public class UserAccidentDocumentAddRequest {

    @NotNull
    private Long carAccidentEntityID;

    @NotNull
    private Date sendDate;

    @NotNull
    private Time sendTime;

    @NotNull
    @NotEmpty
    private String explanationText;
}

File: client/app/src/main/java/com/petrov/vitaliy/caraccidentapp/domain/models/requests/documents/UserAccidentDocumentAddRequest.kt
package com.petrov.vitaliy.caraccidentapp.domain.models.requests.documents

import java.sql.Date
import java.sql.Time

data class UserAccidentDocumentAddRequest(
    val carAccidentEntityID: Long,
    val sendDate: Date,
    val sendTime: Time,
    val explanationText: String
)
